source/infrastructure/pipeline/e2e-tests/utils/vpc-utils.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Construct } from "constructs";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { NagSuppressions } from "cdk-nag";
+
+const testVpcs = new Map<Construct, ec2.Vpc>();
+
+/**
+ * create a default testVPC for a given scope
+ * <p>
+ *   if a vpc has already been defined for this scope, it will be returned instead of creating a new one
+ * </p>
+ */
+export function defaultTestVPC(scope: Construct): ec2.Vpc {
```

Comment:
VPCs are needed for all EC2 and RDS instances. This central utility will allow VPCs to be re-used while still allowing multiple VPCs to be created if in different scopes